scripts/update_plugin.sh

```diff
@@ -16,36 +16,78 @@ cancel() {
 	exit 0
 }
 
-pull_changes() {
-	local plugin="$1"
-	local plugin_path=$(shared_plugin_path "$plugin")
-	cd $plugin_path &&
-		GIT_TERMINAL_PROMPT=0 git pull &&
-		GIT_TERMINAL_PROMPT=0 git submodule update --init --recursive
+update_plugin_git() {
+	local plugin; plugin="$1"
+	local branch; branch=":${1##*:}"
+	local plugin_path; plugin_path=$(shared_plugin_path "$plugin")
+	plugin="${1%$branch}"
+	#update only makes sense when no specific branch/revision is set
+	if [ "${branch}" = ":${1}" ]; then
+		 cd "$plugin_path" &&
+			  GIT_TERMINAL_PROMPT="0" git pull &&
+			  GIT_TERMINAL_PROMPT="0" git submodule update --init --recursive
+	fi
+}
+
+update_plugin_web() {
+	local plugin; plugin="$1"
+	local plugin_name; plugin_name="$(shared_plugin_name "$plugin")"
+	local plugin_path; plugin_path=$(shared_plugin_path "$plugin")
+	(cd "$plugin_path" && rm -rf "${plugin_name}" &&
```

Comment:
The re-downloading of the entire plugin (and the `rm`) make the default behaviour of 'update all' quite expensive, right?